pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -83,14 +91,23 @@ func (s *ServerImpl) PatchAssetScansAssetScanID(ctx echo.Context, assetScanID mo
 	}
 
 	// check that an asset scan with that id exists.
-	_, err = s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
+	existingAssetScan, err := s.dbHandler.AssetScansTable().GetAssetScan(assetScanID, models.GetAssetScansAssetScanIDParams{})
 	if err != nil {
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return sendError(ctx, http.StatusNotFound, fmt.Sprintf("asset scan was not found. assetScanID=%v: %v", assetScanID, err))
 		}
 		return sendError(ctx, http.StatusInternalServerError, fmt.Sprintf("failed to get asset scan. assetScanID=%v: %v", assetScanID, err))
 	}
 
+	// check for valid resource cleanup state transition
+	_, ok := assetScan.GetResourceCleanupStatus()
```

Comment:
```suggestion
	currentResourceCleanupStatus, ok := assetScan.GetResourceCleanupStatus()
	...
			err = currentResourceCleanupStatus.UpdateState(*assetScan.ResourceCleanupStatus.State)
```

Just to avoid updating the `ResourceCleanupStatus` for the current `AssetScan` since this step is only for validation the requested state transition.